Project doesn't build if using v-memo on list rendering since v2.0.18

### Version
Verifiable on v2.0.18 and v2.0.19

### Minimal reproduction
```vue
<script setup lang="ts">
const items = [
  {text: "foo"},
  {text: "bar"}
];
</script>

<template>
  <div v-for="(item, index) in items" v-bind:key="index" v-memo="[item.text]">
    {{ item.text }}
  </div>
</template>
```

OR StackBlitz project: https://stackblitz.com/edit/vitejs-vite-bl8boa

### Steps to reproduce
Run `npm run build` (or just `vue-tsc`) on a project that a component uses list rendering with `v-memo`.

### What is expected?
Minimal valid code to build.

### What is actually happening?
`vue-tsc` fails with the following message:

```
TypeError: Cannot read properties of undefined (reading 'type')
    at generateVFor ([PROJECT PATH]/node_modules/.pnpm/@vue+language-core@2.0.18_typescript@5.4.5/node_modules/@vue/language-core/lib/codegen/template/vFor.js:40:33)
    at generateVFor.next (<anonymous>)
    at generateTemplateChild ([PROJECT PATH]/node_modules/.pnpm/@vue+language-core@2.0.18_typescript@5.4.5/node_modules/@vue/language-core/lib/codegen/template/templateChild.js:94:40)
    at generateTemplateChild.next (<anonymous>)
    at generateTemplateChild ([PROJECT PATH]/node_modules/.pnpm/@vue+language-core@2.0.18_typescript@5.4.5/node_modules/@vue/language-core/lib/codegen/template/templateChild.js:43:20)
    at generateTemplateChild.next (<anonymous>)
    at generateTemplate ([PROJECT PATH]/node_modules/.pnpm/@vue+language-core@2.0.18_typescript@5.4.5/node_modules/@vue/language-core/lib/codegen/template/index.js:20:58)
    at generateTemplate.next (<anonymous>)
    at [PROJECT PATH]/node_modules/.pnpm/@vue+language-core@2.0.18_typescript@5.4.5/node_modules/@vue/language-core/lib/plugins/vue-tsx.js:80:31
    at [PROJECT PATH]/node_modules/.pnpm/computeds@0.0.1/node_modules/computeds/out/computed.js:13:68
src/main.ts:3:17 - error TS2306: File '[PROJECT PATH]/src/App.vue' is not a module.
```

--- a/packages/language-core/lib/codegen/template/vFor.ts
+++ b/packages/language-core/lib/codegen/template/vFor.ts
@@ -54,8 +54,8 @@ export function* generateVFor(
 	for (const argument of node.codegenNode?.children.arguments ?? []) {
 		if (
 			argument.type === CompilerDOM.NodeTypes.JS_FUNCTION_EXPRESSION
-			&& argument.returns.type === CompilerDOM.NodeTypes.VNODE_CALL
-			&& argument.returns.props?.type === CompilerDOM.NodeTypes.JS_OBJECT_EXPRESSION
+			&& argument.returns?.type === CompilerDOM.NodeTypes.VNODE_CALL
+			&& argument.returns?.props?.type === CompilerDOM.NodeTypes.JS_OBJECT_EXPRESSION
 		) {
 			if (argument.returns.tag !== CompilerDOM.FRAGMENT) {
 				isFragment = false;

